code/espurna/thinkspeak.ino

```diff
@@ -95,7 +114,7 @@ void _tspkConfigure() {
         _tspk_enabled = false;
         setSetting("tspkEnabled", 0);
     }
-    if (_tspk_enabled && !_tspk_client) _tspkInitClient();
+    if (_tspk_enabled && !_tspk_client) _tspkInitClient(getSetting("tspkAddress", THINGSPEAK_ADDRESS));
```

Comment:
This can work, but then there should be a note somewhere that url can never change without reboot *or* on initial setup when tspk is still disabled?

Reply:
I haven't seen this note, but, isn't it logical to attach the URL and particularly the host to the client socket to ignore any changes during the request ?
However, I can confirm it, it works !

Reply:
I mean that the `tspkAddress` / `THINGSPEAK_ADDRESS` value is bound to it until we reboot, because client object is never destroyed thus `.address` is never updated again. *Unless*, we also do `_tspk_client.address = URL(getSetting(...))`

Reply:
I update address on post now. That way, there is no need to reboot.